### PR TITLE
Spotify Song analysis

### PR DESCRIPTION
**Pull request by Mattallmighty**

### CODE REVIEW

File: src/store/ui/storeSpotify.tsx [hunk 1]
```diff
@@ -11,7 +166,13 @@ const storeSpotify = () => ({
   spotifyPos: 0 as any,
   spNetworkTime: 1000,
   spAuthenticated: false,
-  spotifyData: {} as any,
+  spotifyData: {
```
Comment (YeonV) [inline src/store/ui/storeSpotify.tsx:169]: @Mattallmighty soooo goood broo

File: src/pages/Integrations/Spotify/SpotifyScreen/SpotifyScreen.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ import SpotifyWidgetPro from '../../../../components/Integrations/Spotify/Widget
 import SpAudioFeatures from '../../../../components/Integrations/Spotify/Widgets/SpotifyWidgetPro/SpAudioFeatures/SpAudioFeatures';
 import SpotifyTriggerTable from '../../../../components/Integrations/Spotify/Widgets/SpotifyWidgetPro/SpTriggerTable';
 import SpPlaylist from '../../../../components/Integrations/Spotify/Widgets/SpotifyWidgetPro/SpPlaylist';
+import Layout from '../../../../components/AudioAnalysis/Layout';
```
Comment (YeonV) [inline src/pages/Integrations/Spotify/SpotifyScreen/SpotifyScreen.tsx:24]: maybe we can have a more descriptive name than layout :)

File: src/components/AudioAnalysis/SectionChart.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import React, { PureComponent } from 'react';
```
Comment (YeonV) [inline src/components/AudioAnalysis/SectionChart.jsx:1]: @Mattallmighty i think this is an old module (jsx) and there is already a tsx duplicate.
shall we remove this file?

File: src/components/AudioAnalysis/PitchSelect.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
```
Comment (YeonV) [inline src/components/AudioAnalysis/PitchSelect.jsx:1]: @Mattallmighty i think this is an old module (jsx) and there is already a tsx duplicate.
shall we remove this file?

**Review by YeonV — COMMENTED** [review]
wonderful

File: src/store/ui/storeSpotify.tsx (hunk 1 above)
Comment (Mattallmighty) [reply src/store/ui/storeSpotify.tsx:169]: Thanks bro!

File: src/pages/Integrations/Spotify/SpotifyScreen/SpotifyScreen.tsx (hunk 2 above)
Comment (Mattallmighty) [reply src/pages/Integrations/Spotify/SpotifyScreen/SpotifyScreen.tsx]: Done

File: src/components/Integrations/Spotify/Widgets/SpotifyWidgetPro/SpTriggerTable.tsx [hunk 5]
```diff
@@ -152,6 +157,25 @@ export default function SpotifyTriggerTable() {
...
+        <FormControl fullWidth disabled>
+          <Select
+            labelId="scene-select-label"
+            value="{sceneId}"
+            label="Scene"
+            onChange={(e: SelectChangeEvent) => {
+              console.log(e);
+            }}
+          >
+            <MenuItem value="sceneId" />
```
Comment (Mattallmighty) [inline src/components/Integrations/Spotify/Widgets/SpotifyWidgetPro/SpTriggerTable.tsx:170]: Add in defined scene, and select dropdown = POST

File: src/components/AudioAnalysis/SpAnalysisChordButtons.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    '&:after': {
+      borderColor: '#ffffff',
+    },
+  },
+  icon: {
+    fill: '#ffffff',
+  },
+}));
+
+const chords: { [key: string]: any } = {
```
Comment (Mattallmighty) [inline src/components/AudioAnalysis/SpAnalysisChordButtons.tsx:28]: When selecting a major chord for example, there is no way to deselect it 🙂 maybe add an option "all"

File: src/components/AudioAnalysis/SpAnalysisChartSize.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+export default function ChartSize({ value, handleSizeSliderChange }: any) {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.root}>
+      <Typography id="input-slider" gutterBottom color="primary">
+        Zoom
+      </Typography>
+      <Grid container spacing={2} alignItems="center">
+        <Grid item xs>
```
Comment (Mattallmighty) [inline src/components/AudioAnalysis/SpAnalysisChartSize.tsx:25]: on my end the graphs feels laggy. it looks amazing but somehow the animations are delaying.
i.e. when changing the zoom

File: src/components/AudioAnalysis/SpAnalysisLayout.tsx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+        return null;
+      });
+    });
+  };
+
+  const { segments, sections } = analysis;
+  // console.log({ analysis });
+
+  return (
+    <Card style={{ color: 'white' }}>
```
Comment (Mattallmighty) [inline src/components/AudioAnalysis/SpAnalysisLayout.tsx:93]: Add collapsalbe accordion, a toggle button or similar to hide it on the first view,
so its not that of an overload for the user

File: src/components/Integrations/Spotify/Widgets/SpotifyWidgetPro/SpTriggerTable.tsx (hunk 5 above)
Comment (Mattallmighty) [reply src/components/Integrations/Spotify/Widgets/SpotifyWidgetPro/SpTriggerTable.tsx]: Done but changed to PUT

File: src/components/AudioAnalysis/SpAnalysisChartSize.tsx (hunk 7 above)
Comment (Mattallmighty) [reply src/components/AudioAnalysis/SpAnalysisChartSize.tsx:25]: Unsure how to resolve this one. Maybe need your expertise on this :)

File: src/components/AudioAnalysis/SpAnalysisChordButtons.tsx (hunk 6 above)
Comment (Mattallmighty) [reply src/components/AudioAnalysis/SpAnalysisChordButtons.tsx:28]: Done

**Pull request closed without merging**